src/types/UiaTextRangeBase.cpp

```diff
@@ -1016,25 +1014,32 @@ try
         _pData->UnlockConsole();
     });
 
+    // We can abstract this movement by moving _start
+    // GH#7342: check if we're past the documentEnd
+    // If so, clamp each endpoint to the end of the document.
+    constexpr auto endpoint = TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start;
+    const auto bufferSize{ _pData->GetTextBuffer().GetSize() };
+    const COORD documentEnd = _getDocumentEnd();
+    if (bufferSize.CompareInBounds(_start, documentEnd, true) > 0)
+    {
+        _start = documentEnd;
+    }

```

Comment:
Man.... we really need to have these things using til::rectangle and have a reasonable clamp method don't we. :(

Reply:
Yeah, I'd love to beef up `til::rectangle`, `til::size`, and `til::point` to be able to clamp and walk through bounds more easily. But unfortunately a lot of that work is blocked by making `Viewport` a `til::rectangle`.

Actually, are we tracking that anywhere? Just about every time I work on a11y, I find new things I want to add to these TIL structs.

Reply:
#4015 is my record of "we still need to kill Viewport"